optimade/server/routers/utils.py

```diff
@@ -319,3 +324,25 @@ def get_single_entry(
         ),
         included=included,
     )
+
+
+def check_params(
+    request: Request, params: Union[EntryListingQueryParams, SingleEntryQueryParams]
+):
+    for param in request.query_params.keys():
+        if not hasattr(params, param):
+            split_param = param.split("_")
+            if param.startswith("_") and len(split_param) > 2:
+                if split_param[1] not in BaseResourceMapper.KNOWN_PROVIDER_PREFIXES:
+                    warn(
+                        f"The Query parameter '{param}' has an unknown provider prefix: '{split_param[1]}'. This query parameter has been ignored.",
+                        UnknownProviderQueryParameter,
+                    )
+                elif split_param[1] in BaseResourceMapper.SUPPORTED_PREFIXES:
+                    raise BadRequest(
+                        detail=f"The query parameter '{param}' has a prefix that is supported by this server, yet the parameter is not known."
+                    )
+            else:
+                raise BadRequest(
+                    detail=f"The query parameter '{param}' is not known by this entry point."
+                )
```

Comment:
I would unify the error messages for the second and third cases (I'm not sure telling the client that the prefix is supported is that much more useful). This should allow the code to be simplified quite a bit (especially wrt. my other comment).